src/sqs.ts

```diff
@@ -92,32 +107,68 @@ export class SQSMessageHandler<Message, Context> {
       // 2. Process all the records.
       context.logger.info({ event }, 'Processing SQS topic message');
 
-      await processWithOrdering(
-        {
-          items: event.Records,
-          // If there is not a MessageGroupId, then we don't care about
-          // the ordering for the event. We can just generate a UUID for the
-          // ordering key.
-          orderBy: (record) => record.attributes.MessageGroupId ?? uuid(),
-          concurrency: this.config.concurrency ?? 5,
-          stopOnError: false,
-        },
-        async (record) => {
-          const messageLogger = context.logger.child({
-            messageId: record.messageId,
-          });
-
-          const parsedMessage = this.config.parseMessage(record.body);
-
-          for (const action of this.messageActions) {
-            await action({ ...context, logger: messageLogger }, parsedMessage);
-          }
-
-          messageLogger.info('Successfully processed message');
-        },
-      );
-
-      context.logger.info('Succesfully processed all messages');
+      const { hasUnprocessedRecords, unprocessedRecords } =
+        await processWithOrdering(
+          {
+            items: event.Records,
+            // If there is not a MessageGroupId, then we don't care about
+            // the ordering for the event. We can just generate a UUID for the
+            // ordering key.
+            orderBy: (record) => record.attributes.MessageGroupId ?? uuid(),
+            concurrency: this.config.concurrency ?? 5,
+            rejectOnError: false,
+          },
+          async (record) => {
+            const messageLogger = context.logger.child({
+              messageId: record.messageId,
+            });
+
+            const parsedMessage = this.config.parseMessage(record.body);
+
+            for (const action of this.messageActions) {
+              await action(
+                { ...context, logger: messageLogger },
+                parsedMessage,
+              );
+            }
+
+            messageLogger.info('Successfully processed message');
+          },
+        );
+
+      if (hasUnprocessedRecords) {
+        context.logger.warn(
+          { unprocessedRecords },
+          'Failed to process all messages',
+        );
+      } else {
+        context.logger.info('Succesfully processed all messages');
+      }
+
+      if (this.config.partialBatching) {
+        // SQS partial batching expects that you return an ordered list of
+        // failures. We map through each group and add them to the batch item
+        // failures in order for each group.
+        const batchItemFailures = Object.entries(unprocessedRecords)
+          .map(([groupId, record]) => {
+            context.logger.warn(
+              { groupId, error: record.error },
+              'Failed to process message group',
+            );
```

Comment:
Two thoughts:

1. We must use `err` for the error to be serialized properly. Also, I'd recommend using `error` instead of `warn`, and updating the log message to be a bit more specific.

```suggestion
            context.logger.error(
              { groupId, err: record.error },
              'Failed to completely process message group',
            );
```

2. Could we update this log object to include _the_ record that failed, and the subsequent unprocessed records in the list?